app/src/main/java/com/github/se/project/ui/components/AcademicSelector.kt

```diff
@@ -3,53 +3,50 @@ package com.github.se.project.ui.components
 import androidx.compose.foundation.background
 import androidx.compose.foundation.border
 import androidx.compose.foundation.clickable
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.fillMaxWidth
-import androidx.compose.foundation.layout.padding
-import androidx.compose.material3.DropdownMenu
-import androidx.compose.material3.DropdownMenuItem
-import androidx.compose.material3.MaterialTheme
-import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.MutableState
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.unit.dp
-import androidx.compose.ui.window.PopupProperties
-import androidx.compose.ui.zIndex
 import com.github.se.project.model.profile.AcademicLevel
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun AcademicSelector(academicLevel: MutableState<AcademicLevel?> = mutableStateOf(null)) {
+fun AcademicSelector(academicLevel: MutableState<AcademicLevel?>, enabled: Boolean = true) {
   val expandedDropdown = remember { mutableStateOf(false) }
+
   Box {
     Text(
         text = academicLevel.value?.name ?: "Academic Level",
         modifier =
             Modifier.fillMaxWidth()
-                .clickable { expandedDropdown.value = !expandedDropdown.value }
-                .background(color = Color.Transparent, shape = MaterialTheme.shapes.small)
-                .border(1.dp, MaterialTheme.colorScheme.outline, MaterialTheme.shapes.small)
+                .clickable(enabled = enabled) { expandedDropdown.value = !expandedDropdown.value }
+                .background(Color.Transparent, shape = MaterialTheme.shapes.small)
+                .border(
+                    1.dp,
+                    if (enabled) MaterialTheme.colorScheme.outline
+                    else MaterialTheme.colorScheme.outline.copy(alpha = 0.4f),
+                    MaterialTheme.shapes.small)
                 .padding(16.dp)
-                .testTag("academicLevelDropdown"),
+                .testTag("academicLevelDropdown")
+                .alpha(if (enabled) 1f else 0.4f), // Visual feedback for disabled state
         style = MaterialTheme.typography.bodyLarge)
 
     DropdownMenu(
-        expanded = expandedDropdown.value,
+        expanded = expandedDropdown.value && enabled,
```

Comment:
Now that we have the Camipro scan feature, it’s good that manual entry is disabled once the scan has already provided the input.